src/stdlib/parse_auditd.rs

```diff
@@ -0,0 +1,432 @@
+
+    if !enrichment.is_empty() {
+        log.insert("enrichment".into(), Value::from(enrichment));
+    }
+
+    Ok(Value::from(log))
+}
+
+impl<'a> TryFrom<AuditValue<'a>> for Value {
+    type Error = Error<'a>;
+    fn try_from(value: AuditValue<'a>) -> Result<Self, Self::Error> {
+        let result = match value {
+            AuditValue::Empty => Value::Null,
+            AuditValue::Str(string, _) => Value::from(string),
+            AuditValue::Number(num) => Value::from(num),
+            AuditValue::List(values) => Value::from(
+                values
+                    .into_iter()
+                    .map(Value::try_from)
+                    .collect::<Result<Vec<_>, _>>()?,
+            ),
+            AuditValue::Owned(string) => Value::from(Bytes::from(string)),
+            // Currently, AuditValue::Map is not generated from the parser (https://github.com/hillu/linux-audit-parser-rs/blob/d8c448c8d8227467b81cd5267790415b8b73f0cb/src/value.rs#L70)
+            // but I think we should use that data type to represent the key-value pairs of the internal (nested) msg field
+            // as depicted here: https://github.com/vectordotdev/vrl/issues/66
+            // For example, in the following linux-audit-parser test https://github.com/hillu/linux-audit-parser-rs/blob/d8c448c8d8227467b81cd5267790415b8b73f0cb/src/test.rs#L168
+            // we see that the msg field is parsed as a string, although it could be key-value pairs
+            AuditValue::Map(entries) => Value::from(
+                entries
+                    .into_iter()
+                    .map(|(key, value)| {
+                        Value::try_from(value).map(|value| (key.to_string().into(), value))
+                    })
+                    .collect::<Result<ObjectMap, _>>()?,
+            ),
+            // There are a few values that `linux-audit-parser` does not return in its parsing
+            // https://github.com/hillu/linux-audit-parser-rs/blob/d8c448c8d8227467b81cd5267790415b8b73f0cb/src/value.rs#L72
+            // We do not plan to support those values, as they are only produced by [laurel](https://github.com/threathunters-io/laurel)
+            // Maybe we should contribute to `linux-audit-parser` to remove the values the parser does not produce, as it
+            // does not have sense in this context.
+            // If those variants are removed, we can remove the error handling of this cases are simplify the code.
+            // Otherwise, we could simply panic in this case
+            unsupported_value => return Err(Error::UnsupportedValue(unsupported_value)),
+        };
+        Ok(result)
+    }
+}
+
+impl From<Number> for Value {
+    fn from(number: Number) -> Self {
+        match number {
+            Number::Dec(decimal) => Value::from(decimal),
+            // TODO: should we store hexadecimals as its integer value or as an hexadecimal string?
+            // TODO: Uppercase hexa or lowercase hex format?
```

Comment:
Good question. Do you happen to know how users would typically want to consume these values or what fields typically contain them? I think the answer to this question likely depends on what sorts of operations users need to do on these values (if any).

Reply:
I don't really know...In the raw log lines this is shown as `a0=55b26d44a6a0` so maybe it makes sense to display it as `"a0": "0x55b26d44a6a0"` in my opinion. But I would love to know what requesters think about this...